packages/core/useWebSocket/index.ts

```diff
@@ -306,6 +312,10 @@ export function useWebSocket<Data = any>(
   const open = () => {
     if (!isClient && !isWorker)
       return
+
+    if (retryTimeout != null)
+      resetRetry()
```

Comment:
i feel like we should be calling this in `close` instead of `open`, just the same as we call `resetHeartbeat` there

regardless of if we reopen later, closing should stop the timer

Reply:
I want to solve the issue where the auto-reconnect timeout might trigger unnecessarily when manually opening a new connection.

When a connection drops, our autoReconnect sets up a retryTimeout for delayed reconnection. If the user manually calls open() before this timeout triggers, we need to clear this timeout first - otherwise it would trigger another connection attempt even after we've already established a new connection manually.

Maybe there's a better way to handle this, but this is my current approach. Thanks for your review :>

Reply:
`open` calls `close` before doing anything else, so it will also result in calling this additional logic if you put it there

Reply:
Sorry, I'm stuck. Could you help me test this [reproduction](https://stackblitz.com/edit/vitejs-vite-pqu1gc?file=src%2FApp.vue)? I tried to move it, but it doesn't seem to work 🥲

Reply:
if i do the following in a test, it fails before the change and passes after:

1. open the socket (with `autoReconnect: true`)
2. close the socket (by causing the `WebSocket` to emit a `close` rather than by calling `close()` ourselves)
3. call `open()` (**at this point, the reconnect timer is queued already)

it seems to work

i just changed `close` to do this:

```ts
resetHeartbeat()
resetRetry()
heartbeatPause?.()
```

you can see in `open`, it calls `close` first thing:

```ts
const open = () => {
  if (!isClient && !isWorker)
    return
  close() // CALLED HERE
  // ...
}
```

so this change is doing exactly what your original code did (for this particular test case)

it'll fix another case if we do it this way too